src/main/java/org/testng/annotations/NewInstancePerMethod.java

```diff
@@ -0,0 +1,16 @@
+package org.testng.annotations;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.TYPE;
+
+/**
+ * Marking a class with this tells TestNG to use a new instance of the class per test method.  You can also do this
+ * globally with a system property (testng.newInstancePerMethod) or environment variable
+ * "TESTNG_NEW_INSTANCE_PER_METHOD".
+ */
+@Retention(java.lang.annotation.RetentionPolicy.RUNTIME)
+@Target(TYPE)
+public @interface NewInstancePerMethod {
```

Comment:
Not sure if you would expect this feature to work with inheritance as well. If yes Should we add `@java.lang.annotation.Inherited` ? If this observation is true, please help add a test that verifies this as well.